### PR TITLE
Auto uses wormholes before the End Game

### PR DESCRIPTION
**Pull request by GreatOldOne**
Time before End Game is configurable
Also adds an optional argument to tryUsingItem to check if item is
active in lane
isNearGame() function can be used to determine if the game is near the
configurable end

### CODE REVIEW

**Comment by xPaw** [comment]
Why aren't you using game time instead of real time?

File: autoPlay.user.js [hunk 1]
```diff
@@ -251,7 +258,23 @@ function disableParticles() {
 	}
 }
 
+function updateCurrentTime() {
+    cHours = cTime.getUTCHours();
```
Comment (xPaw) [inline autoPlay.user.js]: This will always return the same value as `cTime` itself is never re-initialized.

File: autoPlay.user.js [hunk 2]
```diff
@@ -1251,13 +1292,18 @@ function tryUsingAbility(abilityId) {
...
-		return false;
-	}
-
-	triggerItem(itemId);
-	return true;
+function tryUsingItem(itemId, checkInLane) {
+    if (!canUseItem(itemId)) {
+        return false;
+    }
+    if (typeof checkInLane === "undefined") {
```
Comment (xPaw) [inline autoPlay.user.js]: This `if` is entirely pointless.

File: autoPlay.user.js (hunk 1 above)
Comment (GreatOldOne) [inline autoPlay.user.js]: Because End Game always occurs at 15 GMT regardless of game time.

Comment (xPaw) [inline autoPlay.user.js]: You want this function to return current time, don't you? If so, this isn't going to work.

**Comment by xPaw** [comment]
Also please use tabs instead of spaces to keep it consistent.

File: autoPlay.user.js (hunk 1 above)
Comment (GreatOldOne) [inline autoPlay.user.js]: Yeah, We need reintialize it every time updateCurrentTime is called.
I'll fix it.

File: autoPlay.user.js (hunk 2 above)
Comment (GreatOldOne) [inline autoPlay.user.js]: It's not. It's an optional argument. Otherwise I would have to change every single call to tryUsingItem out there.

Comment (xPaw) [inline autoPlay.user.js]: Next if will only pass if `checkInLane` is set to something positive. It won't if it's false or undefined.

**Comment by GreatOldOne** [comment]
All my text editors are configured to 4 whitespaces on tab. Couldn't you tabify everything with a single click and merge? Thanks.

**Comment by xPaw** [comment]
We have an `.editorconfig` in the repository too. And I'm not going to do it for you because that's an extra commit.

File: autoPlay.user.js (hunk 2 above)
Comment (GreatOldOne) [inline autoPlay.user.js]: Yeah, that's why i pass true to tryUsingItem on useWormholeIfRelevant. It's only false if you pass false or don't define it at all.

Comment (xPaw) [inline autoPlay.user.js]: Correct. That's why I said your "is undefined -> set to false" is not needed.

Comment (GreatOldOne) [inline autoPlay.user.js]: Yeah, sure. I thought it was more of a good practice to state that checkInLane is optional and defaults to false.
I can remove it if you wish though.

Comment (xPaw) [inline autoPlay.user.js]: It's unnecessary code, remove it.

**Comment by GreatOldOne** [comment]
All done as requested!

**Comment by xPaw** [comment]
Looks good. The only thing I would do is move all the time stuff inside `isNearEndGame`, as there is no need to keep all these time variables global.

**Comment by GreatOldOne** [comment]
cMins and cHours need to be globals for updateCurrentTime and isNearEndGame to access.
As for cTime it can be moved to local, but it could be left this way for other functions to access the whole timestamp if necessary.

**Comment by xPaw** [comment]
There is no need for `updateCurrentTime` to exist, just keep that functionality inside `isNearEndGame`.

**Comment by GreatOldOne** [comment]
But what if we create another function that wants to check this data ourside isNearEndGame?
Duplicate?

**Comment by xPaw** [comment]
In that case we can rename `useWormholeIfRelevant` and add extra stuff there, alternatively `isNearEndGame` can accept minsLeft argument. Anyway, there's no reason to keep time globally.

**Comment by GreatOldOne** [comment]
Done!

**Comment by xPaw** [comment]
Looks fine now, i`ll wait for @chauffer or @wchill to review this.

**Comment by GreatOldOne** [comment]
We should probably pass checkInLane true to decrease cooldown as well, since it doesn't stack.

**Comment by iakat** [comment]
@GreatOldOne That'd be for another PR.
Looks good! :ship:

**Pull request merged**